lib/utils.js

```diff
@@ -0,0 +1,84 @@
+/*!
+ * body-parser
+ * Copyright(c) 2014-2015 Douglas Christopher Wilson
+ * MIT Licensed
+ */
+
+'use strict'
+
+/**
+ * Module dependencies.
+ */
+
+var bytes = require('bytes')
+var contentType = require('content-type')
+var typeis = require('type-is')
+
+/**
+ * Module exports.
+ */
+
+module.exports = {
+  getCharset,
+  normalizeOptions
+}
+
+/**
+ * Get the charset of a request.
+ *
+ * @param {object} req
+ * @api private
+ */
+
+function getCharset (req) {
+  try {
+    return (contentType.parse(req).parameters.charset || '').toLowerCase()
+  } catch (e) {
+    return undefined
+  }
+}
+
+/**
+ * Get the simple type checker.
+ *
+ * @param {string} type
+ * @return {function}
+ */
+
+function typeChecker (type) {
+  return function checkType (req) {
+    return Boolean(typeis(req, type))
+  }
+}
+
+/**
+ * Normalizes the common options for all parsers.
+ *
+ * @param {object} options
+ * @param {string} defaultType
+ * @returns {object}
+ */
+function normalizeOptions (options, defaultType) {
```

Comment:
We could throw if `defaultType` is not provided

Previously it was implicitly required in that all parsers defined a fallback, but the current approach will accept undefined here silently and make the parser match against any Content-Type header provided in a req. It's unlikely that is expected or desired behavior on the part of a maintainer creating a new parser.

I put it up top because we are verifying internal use here, not the normalized value after considering the user input.

```suggestion
function normalizeOptions (options, defaultType) {
  if (!defaultType) {
    // Parsers must define a default content type or function
    throw new TypeError('defaultType must be provided')
  }
```